### DISCUSSION TITLE
Nuget package references to different System.Buffers than actual dll inside it

### DISCUSSION THREAD

**Discussion by turoj** [rabbitmq/rabbitmq-dotnet-client#1194] [General]
**What is affected by this:**
This seems to be case in both in main and 6.x branches.
Also package in nuget.org affected by this.

**Problem:**
RabbitMQ.Client DLL references to
System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51

While nuget has reference to package System.Memory 4.5.4, which references System.Buffers 4.5.1 and that includes version
System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51

Previous version of System.Memory would point to correct version of System.Buffers.

**What it causes:**
This seems to cause problems only in some projects that use RabbitMQ.Client, but at least one case is that during starting listener it will just throw exception:

_Inner EXCEPTION (1): connection.start was never received, likely due to a network timeout
TYPE: System.IO.IOException 
STACK TRACE:    at RabbitMQ.Client.Framing.Impl.Connection.StartAndTune()
   at RabbitMQ.Client.Framing.Impl.Connection.Open(Boolean insist)
   at RabbitMQ.Client.Framing.Impl.Connection..ctor(IConnectionFactory factory, Boolean insist, IFrameHandler frameHandler, String clientProvidedName)
   at RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(IFrameHandler fh)
   at RabbitMQ.Client.ConnectionFactory.CreateConnection(IEndpointResolver endpointResolver, String clientProvidedName)_

On server side it just looks like:

_[info] <0.766.0> accepting AMQP connection <0.766.0> (172.17.0.1:47346 -> 172.17.0.2:5672)
[warning] <0.766.0> closing AMQP connection <0.766.0> (172.17.0.1:47346 -> 172.17.0.2:5672):
[warning] <0.766.0> client unexpectedly closed TCP connection_

**Fix:**
Fix is to use same version of System.Buffers library than what RabbitMQ.Client DLL references to.

**Comment by michaelklishin** [comment]
I could not find any direct dependencies on on that version. So it must be multiple dependencies of ours depending on different versions of `System.Buffers`, a classic transient dependency collision.


@turoj you are welcome to track down where those dependencies come from and submit a suitable PR.
We might be forced to change certain versions of other dependencies to resolve this (find a common `System.Buffers` version.

**Comment by pabermod** [comment]
Hi, i'm having the same issue, in a project where another libraries uses different versions of System.Threading.Channels, System.Buffers, System.Runtime.CompilerServices.Unsafe, etc... I'm getting this error (the righ assembly binding redirects are defined)

In a clean console app this problem is not happening

**Reply by pabermod** [reply]
Turns out I was missing some redirects... But its quite difficult to know this is the source of the error
